### PR TITLE
Added badges

### PR DESCRIPTION
**Pull request by larsenv**
WEW

Added the badges for the profiles. It works universally on any server!

### CODE REVIEW

**Comment by Seriell** [comment]
I'll test it tomorrow and merge if it works to my satisfaction

nobody else merge it while I'm sleeping ty <3

**Comment by larsenv** [comment]
I noticed that the badges are kinda fucked up. Like if I do !code lookup for Procyon it still shows my badges and not hers. I need to take a break from this for a bit, please make event properly handle this if you can.

Also, I commented out verified as it needs to be fixed.

Please help, thanks!

**Comment by Seriell** [comment]
I'll check it tomorrow

File: modules/codes.rb [hunk 1]
```diff
@@ -113,7 +113,17 @@ def self.modify_codes(user_id, args, option)
                         '3ds'.to_sym => '<:New3DSXL:287651327763283968> **3DSs**:',
                         :nnid => '<:NintendoNetworkID:287655797104836608> **Nintendo Network IDs**:',
                         :switch => '<:Switch:287652338791874560> **Switches**:',
-                        :game => '🎮 **Games**:'
+                        :game => '🎮 **Games**:',
```
Comment (spotlightishere) [inline modules/codes.rb]: Not really sure why you'd add a comma, if I remember Ruby gets angry over that. If not ignore this

File: modules/helper.rb [hunk 2]
```diff
@@ -113,6 +121,21 @@ def self.is_verified?(event, other_user = nil)
       return is_xxx_role?(event, 'vfd', 'Verified', true, user)
     end
 
+    def self.get_roles(event)
+      roles = []
```
Comment (spotlightishere) [inline modules/helper.rb]: I'd highly advise against doing this in the first place. It's slow enough to check by events which tbh I should've never done in the first place. Perhaps a better idea would be to cache the roles somehow for a user; code lookup is slow enough as it is. I was mulling over ideas on how to improve the actual code part earlier, perhaps I should start thinking on badges as well.

File: modules/helper.rb [hunk 3]
```diff
@@ -95,10 +95,18 @@ def self.is_helper?(event)
...
+      return is_xxx_role?(event, 'don', 'Donators', false)
+    end
+
     def self.is_server_admin?(event)
       return is_xxx_role?(event, 'adm', 'Admin', false)
     end
 
+    def self.is_translator?(event)
+      return is_xxx_role?(event, 'trn', 'Translator', false)
+    end
```
Comment (spotlightishere) [inline modules/helper.rb]: Isn't it Translators? Ruby takes the role names very strictly

File: modules/helper.rb [hunk 4]
```diff
@@ -113,6 +121,21 @@ def self.is_verified?(event, other_user = nil)
...
+
+      # I'm sure there's a better way to do this. Please fix it.
+
+      if is_developer?(event) then roles << "dev" end
+      if is_moderator?(event) then roles << "mod" end
+      if is_helper?(event) then roles << "hlp" end
+      if is_donator?(event) then roles << "don" end
+      if is_server_admin?(event) then roles << "adm" end
+      if is_translator?(event) then roles << "trn" end
+      # if is_verified?(event) then roles << "vfd" end
```
Comment (spotlightishere) [inline modules/helper.rb]: I noticed you mentioned not doing verified -- I'd recommend you remove this. It compares user, not event and was to check if another user than the current one was verified. The entire function should probably be removed.

Comment (spotlightishere) [inline modules/helper.rb]: Also, verified users shouldn't have badges tbh.

**Comment by spotlightishere** [comment]
One last thing I forgot to mention in my review, please check what server this is and compare it to RC24 so that we don't have phone badges. @Seriell is there any chance on the config you could have a home server ID option, and remove ID references in Morpher and birthdays and such?

File: modules/helper.rb (hunk 4 above)
Comment (ArtutoGamer) [inline modules/helper.rb]: Agree with spot

**Pull request merged**